datastore-design/go-grpc-demo/examples/clients/python/client_knmi_param.py

```diff
@@ -0,0 +1,107 @@
+import datastore_pb2_grpc as dstore_grpc
+import grpc
+import pandas as pd
+import xarray as xr
+from parameters import knmi_parameter_names
+from google.protobuf.timestamp_pb2 import Timestamp
+
+
+def netcdf_file_to_requests(file_path: Path | str) -> Tuple[List, List]:
+    time_series_request_messages = []
+    observation_request_messages = []
+    ts_id = 1
+
+    with xr.open_dataset(file_path, engine="netcdf4", chunks=None) as file:  # chunks=None to disable dask
+        for param_id in knmi_parameter_names:
+            ts_observations = []
+
+            param_file = file[param_id]
+            for station_id, latitude, longitude, height in zip(
+                file["station"].values, file["lat"].values[0], file["lon"].values[0], file["height"].values[0]
+            ):
+                tsMData = dstore.TSMetadata(
+                    station_id=station_id,
+                    param_id=param_id,
+                    lat=latitude,
+                    lon=longitude,
+                    other1=param_file.name,
+                    other2=param_file.long_name,
+                    other3="value3",
+                )
+                request = dstore.AddTSRequest(
+                    id=ts_id,
+                    metadata=tsMData,
+                )
+
+                time_series_request_messages.append(request)
+
+                station_slice = param_file.sel(station=station_id)
+
+                observations = []
+                for time, obs_value in zip(
+                    pd.to_datetime(station_slice["time"].data).to_pydatetime(), station_slice.data
+                ):
+                    ts = Timestamp()
+                    ts.FromDatetime(time)
+                    observations.append(
+                        dstore.Observation(
+                            time=ts,
+                            value=obs_value,
+                            metadata=dstore.ObsMetadata(field1="KNMI", field2="Royal Dutch Meteorological Institute"),
+                        )
+                    )
+
+                ts_observations.append(dstore.TSObservations(tsid=ts_id, obs=observations))
+                ts_id += 1
+
+            request = dstore.PutObsRequest(tsobs=ts_observations)
+            observation_request_messages.append(request)
+
+    return time_series_request_messages, observation_request_messages
+
+
+def insert_data(time_series_request_messages: List, observation_request_messages: List):
+    workers = int(cpu_count())
+
+    with grpc.insecure_channel(f"{os.getenv('DSHOST', 'localhost')}:{os.getenv('DSPORT', '50050')}") as channel:
+        client = dstore_grpc.DatastoreStub(channel=channel)
+
+        print(f"Inserting {len(time_series_request_messages)} time series requests.")
+        time_series_insert_start = perf_counter()
+        with concurrent.futures.ThreadPoolExecutor(max_workers=workers) as executor:
+            executor.map(client.AddTimeSeries, time_series_request_messages)
```

Comment:
Check for errors. It will now look really fast if for example the gRPC server is down.